### PR TITLE
Remove unnecessary Mono.Security references from corlib.

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: mcs/class/corlib/System.Security.Policy/Evidence.cs [hunk 1]
```diff
@@ -260,16 +260,14 @@ static internal Evidence GetDefaultHostEvidence (Assembly a)
...
-					if (x509.GetHashCode () != 0) {
-						e.AddHost (new Publisher (x509));
-					}
+				try {
+					X509Certificate x509 = X509Certificate.CreateFromSignedFile (a.Location);
+					e.AddHost (new Publisher (x509));
+				}
+				catch (CryptographicException) {
 				}
 			}
```
Comment (filipnavara) [inline mcs/class/corlib/System.Security.Policy/Evidence.cs]: The comment about X509Certificate.CreateFromSignedFile is totally bogus. AuthenticodeDeformatter checks the signature and the result is ignored. X509Certificate.CreateFromSignedFile is supposed to only load the certificate, but not validate its chain, so in theory it's a weaker verification.

X509Certificate.CreateFromSignedFile is currently non-working in master, in the same way as it is broken in CoreFX. That's a separate issue though.

**Comment by marek-safar** [comment]
This fails to build for all profiles

**Comment by filipnavara** [comment]
@marek-safar Working on it, but it actually fails for monotouch_watch only now.

**Comment by filipnavara** [comment]
@monojenkins build Linux x64

**Comment by filipnavara** [comment]
The remaining build failures are unrelated to this PR.

**Comment by filipnavara** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins commit csproj

File: mcs/class/corlib/System.Security.Policy/Evidence.cs (hunk 1 above)
Comment (baulig) [reply mcs/class/corlib/System.Security.Policy/Evidence.cs]: Let's submit that to CoreFX as we're now using their implementation.

Comment (filipnavara) [reply mcs/class/corlib/System.Security.Policy/Evidence.cs]: @baulig I have a patch queue for CoreFX. It's a bit overflowing due to all the vacations of other people and some blocking issues, but I am aware of this and I do plan to fix it.

**Pull request merged**